### ISSUE TITLE
Unable to save preferences - no save button

### ISSUE THREAD

**Issue by philhawthorne** [andrew-codechimp/ha-menu#24]
Great little app idea!

Unfortunately I'm unable to save the preferences, as thee are no save buttons. I've also tried just pressing enter, but that doesn't work either.

![image](https://user-images.githubusercontent.com/716711/69311819-669b7500-0c81-11ea-809a-b696c671c81b.png)

![image](https://user-images.githubusercontent.com/716711/69311833-72873700-0c81-11ea-9f4c-4a5d63c31042.png)

Running MacOS 10.14.6

**Comment by andrew-codechimp** [comment]
The preferences are automatically saved when you hit the red close button (typical on MacOS)

If you're not seeing switches here's a couple of thoughts

- Typically you need a port number on the end of your IP, e.g. http://192.168.1.123:8123
- The token must be copy/pasted exactly from HA's long lived tokens screen, ensure no leading/trailing spaces are added.
- When you click to open HA Menu it will try and get switches using the current settings.  Only entities starting with switch. (not lights).  If you don't have any switches and want to map a light/scene to a switch create a template switch within HA config.

Error reporting is non-existent at the moment, I will try and add some status reporting to help diagnose connection issues soon.

**Comment by philhawthorne** [comment]
Thanks Andrew.

So I think what's happening is I am opening the preferences, and then using CMD + Tab to switch windows to copy the token. 

However I can't CMD + Tab back to the preferences. So I open a new preferences window. When I click the red X, I am then shown the original preferences window which is now showing the old data. The only way to remove this window is to use the red X, which then saves the preferences again now with my new settings gone.

I've been able get the settings to save now, however still nothing showing up for me. I've tried with both my Nabu Casa URL, and my local URL. I have many `switch` entities, so should have at least something show up. Unless I have too many for it to handle (which could be a possibility too I'm guessing)

**Comment by andrew-codechimp** [comment]
CMD Tab is a pain when an app doesn't have an app icon.  I usually minimize windows to see them.

Just spent the morning adding a lot of error handling to give an indication of common issues, these appear where the switches should be as per screenshot.

Go get V1.5 from Releases, unzip and replace the app in your application folder (quit HA Menu first)
https://github.com/andrew-codechimp/ha-menu/releases/tag/V1.5

Don't use Nabu Casa myself but pretty sure you should use your internal IP. 

<img width="312" alt="Screenshot 2019-11-21 at 12 57 40" src="https://user-images.githubusercontent.com/1849731/69340451-73a76c80-0c5f-11ea-9793-0e0bc08db862.png">

**Comment by andrew-codechimp** [comment]
And if all else fails, you can execute a curl statement from terminal as follows
curl -i http://YOURIP:8123/api/states -H "Authorization: Bearer YOURTOKEN"
Replace the YOURIP and YOURTOKEN
That should get your full server state, starting with HTTP/1.1 200 OK 
Probably a lot of sensitive stuff in that so don't post it but it confirms the basics are working.

**Comment by philhawthorne** [comment]
Looks like now the menu completely blanks out when data comes through.

With the default group `all_switches` set, I briefly see the preferences menu and then this happens

![image](https://user-images.githubusercontent.com/716711/69384499-95acf780-0d10-11ea-9aaa-0025e440fbf0.png)

Making that curl request with the same token from HA-Menu works. The payload is quite large. Not sure if this could be the cause?

```
HTTP/1.1 200 OK
Content-Type: application/json
Content-Length: 465176
Date: Thu, 21 Nov 2019 23:13:59 GMT
Server: Python/3.7 aiohttp/3.6.1
```

**Comment by andrew-codechimp** [comment]
OK, that's ruled out the connection issue at least.
I actually have two instances of HA (long story) but I'm able to replicate this on one of them now so will investigate.  It does appear to be an issue with the payload, something is causing the parser to error so need to code that more defensively.

**Comment by andrew-codechimp** [comment]
I found that adding a trailing slash in server causes this.  
Added a fix in version 1.6, hopefully that's all it is.

**Comment by andrew-codechimp** [comment]
Closing this issue as the original problem of non modal preferences resolved in release 1.7
Issue #25 is tracking json parser issue.
